### PR TITLE
[SDK-3559] Default error handler

### PR DESCRIPTION
**Pull request by adamjmcgrath**
- [ ] All new/changed/fixed functionality is covered by tests (or N/A)
- [ ] I have added documentation for all new/changed functionality (or N/A)

### 📋 Changes

Add a default `onError` handler and set the default handler to just print the status (and log the error)

### 🎯 Testing

```bash
$ npm run start:kitchen-sink
# Visit http://localhost:api/auth/callback?error=foo&error_description=bar
```

### CODE REVIEW

File: tests/handlers/logout.test.ts [hunk 1]
```diff
@@ -99,11 +89,4 @@ describe('logout handler', () => {
       Path: '/'
     });
   });
-
-  test('should escape html in error message', async () => {
```
Comment (adamjmcgrath) [inline tests/handlers/logout.test.ts:103]: These tests only really made sense for the callback handler

**Comment by Widcket** [comment]
CI is failing.

File: V2_MIGRATION_GUIDE.md [hunk 2]
```diff
@@ -113,3 +114,45 @@ Previously the ID token would be stored in the session cookie, making the cookie
...
+### After
+
+```js
+export default handleAuth({
+  onError(req, res, error) {
+    errorLogger(error);
+    // You can finish the response yourself if you want to customise
+    // the status code or redirect the user
+    // res.writeHead(302, {
+    //     Location: '/cusotm-error-page'
```
Comment (Widcket) [inline V2_MIGRATION_GUIDE.md:153]:
```suggestion
    //     Location: '/custom-error-page'
```

File: V2_MIGRATION_GUIDE.md [hunk 3]
```diff
@@ -113,3 +114,45 @@ Previously the ID token would be stored in the session cookie, making the cookie
...
+});
+```
+
+### After
+
+```js
+export default handleAuth({
+  onError(req, res, error) {
+    errorLogger(error);
+    // You can finish the response yourself if you want to customise
```
Comment (Widcket) [inline V2_MIGRATION_GUIDE.md:150]:
```suggestion
    // You can finish the response yourself if you want to customize
```

**Review by Widcket — APPROVED** [review]

**Pull request merged**